### PR TITLE
#1722 Solved Nested SubQueries Working

### PR DESCRIPTION
**Pull request by gargayush18**
# Summary of changes

# Checklist

- [x] Checked that there aren't other open pull requests for
  the same change.
- [ ] Added tests for any changes that have a runtime impact.
- [x] Checked that all tests and type checking passes.
- For changes that have a potential impact on users of this project:
  - [ ] Updated relevant documentation to avoid inaccuracies.
  - [ ] Considered adding additional documentation.
  - [ ] Considered adding an example in `./examples` for new features.
  - [ ] Considered updating our changelog (`CHANGELOG.md`).
- [x] Considered granting [push permissions to the PR branch](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork),
  so maintainers can fix minor issues and keep your PR up to date.

### CODE REVIEW

**Comment by gargayush18** [comment]
pre-commit.ci autofix

**Comment by aucampia** [comment]
This will need tests and you should remove the commented code also.

File: rdflib/plugins/sparql/processor.py [hunk 1]
```diff
@@ -23,15 +23,6 @@ def prepareQuery(queryString, initNs={}, base=None) -> Query:
...
 
-def prepareUpdate(updateString, initNs={}, base=None):
-    """
-    Parse and translate a SPARQL Update
-    """
-    ret = translateUpdate(parseUpdate(updateString), base, initNs)
-    ret._original_args = (updateString, initNs, base)
-    return ret
-
-
```
Comment (aucampia) [inline rdflib/plugins/sparql/processor.py:34]: Why are you removing this?

Comment (gargayush18) [reply rdflib/plugins/sparql/processor.py:34]: Okay, I will update the changes suggested . I will add the function back again and also update the formatting of the code

File: rdflib/plugins/sparql/processor.py [hunk 2]
```diff
@@ -66,16 +57,116 @@ class SPARQLProcessor(Processor):
...
 
+    def check(s):
+        pass
+
+    def Describe(self, strOrQuery):
+        pass
+
+    def clean_query(self, strOrQuery):
+
+        temp = strOrQuery.replace("SELECT *", "SELECT **SELECT *").replace("\n", "")
```
Comment (aucampia) [inline rdflib/plugins/sparql/processor.py:68]: This seems to be based on string manipulation which is not ideal and I don't think strOrQuery will always be a sting. It should be done on the AST instead, and I'm also not sure I understand the reasoning here, we should not clean the query to handle nested subqueries, we should just handle them.

**Review by aucampia — CHANGES_REQUESTED** [review]
See previous comments.

**Comment by aucampia** [comment]
Please rebase this branch onto master or merge master into this branch.

**Comment by aucampia** [comment]
Closing this PR due to inactivity, feel free to open it again if you continue working on it again.

**Pull request closed without merging**